services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt

```diff
@@ -269,6 +276,15 @@ class ApplicationController(
         }
     }
 
+    fun checkApplicationPermission(hankeTunnus: String?, userId: String) {
+        hankeTunnus.let { tunnus ->
+            hankeService.getHankeId(tunnus).let { id ->
+                if (id == null || !permissionService.hasPermission(id, userId, EDIT_APPLICATIONS))
+                    throw HankeNotFoundException(hankeTunnus)
+            }
```

Comment:
Using control structures increases the cognitive load required to understand the code, so they shouldn't be used unnecessarily.

```suggestion
            val hankeId = hankeService.getHankeId(hankeTunnus)
            if (hankeId == null || !permissionService.hasPermission(hankeId, userId, EDIT_APPLICATIONS))
                throw HankeNotFoundException(hankeTunnus)
```